diff --git a/tclpkg/gv/gv.cpp b/tclpkg/gv/gv.cpp
index 5f6d03f69..4222e829f 100644
--- a/tclpkg/gv/gv.cpp
+++ b/tclpkg/gv/gv.cpp
@@ -156,7 +156,7 @@ char *setv(Agraph_t *g, char *attr, char *val) {
     return nullptr;
   Agsym_t *a = agfindattr(agroot(g), attr);
   if (!a)
-    a = agraphattr(g->root, attr, emptystring);
+    a = agraphattr(agroot(g), attr, emptystring);
   myagxset(g, a, val);
   return val;
 }
@@ -303,9 +303,12 @@ Agnode_t *tailof(Agedge_t *e) {
 }
 
 Agraph_t *graphof(Agraph_t *g) {
-  if (!g || g == g->root)
+  if (!g)
     return nullptr;
-  return agroot(g);
+  Agraph_t *root = agroot(g);
+  if (g == root)
+    return nullptr;
+  return root;
 }
 
 Agraph_t *graphof(Agedge_t *e) {
